src/screens/ImportWalletScreen.tsx

```diff
@@ -19,88 +18,228 @@ import {
+  HDSegwitP2SHArWallet,
   HDSegwitBech32Wallet,
+  HDSegwitP2SHAirWallet,
 } from '../../class';
 
 const i18n = require('../../loc');
 
 interface Props {
   navigation: CompositeNavigationProp<
-    StackNavigationProp<MainTabNavigatorParams, Route.ContactList>,
-    CompositeNavigationProp<
-      StackNavigationProp<RootStackParams, Route.DeleteContact>,
-      StackNavigationProp<MainCardStackNavigatorParams, Route.ImportWallet>
-    >
+    StackNavigationProp<MainTabNavigatorParams, Route.Dashboard>,
+    StackNavigationProp<MainCardStackNavigatorParams, Route.ImportWallet>
   >;
+  route: RouteProp<MainCardStackNavigatorParams, Route.ImportWallet>;
   loadWallets: () => Promise<WalletsActionType>;
 }
 
-export const ImportWalletScreen = (props: Props) => {
-  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
-  const [text, setText] = useState('');
-  const [validationError, setValidationError] = useState('');
+interface State {
+  isButtonDisabled: boolean;
+  text: string;
+  label: string;
+  validationError: string;
+}
+
+export class ImportWalletScreen extends PureComponent<Props, State> {
+  state = {
+    isButtonDisabled: true,
+    text: '',
+    label: '',
+    validationError: '',
+  };
 
-  const showErrorMessageScreen = () =>
+  showErrorMessageScreen = () =>
     CreateMessage({
       title: i18n.message.somethingWentWrong,
       description: i18n.message.somethingWentWrongWhileCreatingWallet,
       type: MessageType.error,
       buttonProps: {
         title: i18n.message.returnToDashboard,
-        onPress: () => props.navigation.navigate(Route.Dashboard),
+        onPress: () => this.props.navigation.navigate(Route.ImportWalletChooseType),
       },
     });
 
-  const showSuccessImportMessageScreen = () =>
+  showSuccessImportMessageScreen = () =>
     CreateMessage({
       title: i18n.message.success,
       description: i18n.message.successfullWalletImport,
       type: MessageType.success,
       buttonProps: {
         title: i18n.message.returnToDashboard,
-        onPress: () => props.navigation.navigate(Route.Dashboard),
+        onPress: () => this.props.navigation.navigate(Route.Dashboard),
       },
     });
 
-  const onImportButtonPress = async () => {
+  onImportButtonPress = async () => {
     Keyboard.dismiss();
     CreateMessage({
       title: i18n.message.creatingWallet,
       description: i18n.message.creatingWalletDescription,
       type: MessageType.processingState,
-      asyncTask: () => importMnemonic(text),
+      asyncTask: () => this.importMnemonic(this.state.text),
     });
   };
 
-  const onChangeText = (mnemonic: string) => {
-    setText(mnemonic);
-    setValidationError('');
-    if (isButtonDisabled !== (mnemonic.length === 0)) {
-      setIsButtonDisabled(!isButtonDisabled);
+  onChangeText = (mnemonic: string) => {
+    this.setState({ text: mnemonic, validationError: '' });
+    if (this.state.isButtonDisabled !== (mnemonic.length === 0)) {
```

Comment:
I would remove the isButtonDisabled variable and make function like canSubmit() => {
const { text, validationError} = this.state
return !validationError && !text.trim();
}
this is more flexible solution